### PR TITLE
Wipe for failed init

### PR DESCRIPTION
**Pull request by tegansiobhan**
Re-opening of PR 449 as a draft pull request. Updates proof of aws_array_list_init_dynamic and aws_priority_queue_init_dynamic to check for wiped structures in failure cases.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tegansiobhan** [comment]
@danielsn @ColdenCullen @feliperodri

File: source/priority_queue.c [hunk 1]
```diff
@@ -263,6 +265,16 @@ bool aws_priority_queue_is_valid(const struct aws_priority_queue *const queue) {
     return pred_is_valid && container_is_valid && backpointers_valid;
 }
 
+bool aws_priority_queue_is_wiped(const struct aws_priority_queue *const queue) {
```
Comment (ColdenCullen) [inline source/priority_queue.c]: I think we should figure out what we're doing with #453 before we add any more of these.

Comment (feliperodri) [reply source/priority_queue.c]: +1

**Comment by graebm** [comment]
https://github.com/awslabs/aws-c-common/pull/453 is merged

File: .cbmc-batch/jobs/aws_priority_queue_init_dynamic/aws_priority_queue_init_dynamic_harness.c [hunk 2]
```diff
@@ -37,6 +37,8 @@ void aws_priority_queue_init_dynamic_harness() {
 
     /* perform operation under verification */
     uint8_t *raw_array = bounded_malloc(len);
+
+    /* Store which parameter choice is made for queue to confirm postconditions */
```
Comment (danielsn) [inline .cbmc-batch/jobs/aws_priority_queue_init_dynamic/aws_priority_queue_init_dynamic_harness.c]: Not sure what this comment means

File: include/aws/common/priority_queue.h [hunk 3]
```diff
@@ -117,6 +117,12 @@ bool aws_priority_queue_backpointers_valid(const struct aws_priority_queue *cons
 AWS_COMMON_API
 bool aws_priority_queue_is_valid(const struct aws_priority_queue *const queue);
 
+/**
+ * Set of properties of a wiped aws_priority_queue.
+ */
+AWS_COMMON_API
+bool aws_priority_queue_is_wiped(const struct aws_priority_queue *const queue);
```
Comment (danielsn) [inline include/aws/common/priority_queue.h]: Change to use the macros from https://github.com/awslabs/aws-c-common/pull/453

File: source/priority_queue.c [hunk 4]
```diff
@@ -169,6 +169,8 @@ int aws_priority_queue_init_dynamic(
     int ret = aws_array_list_init_dynamic(&queue->container, alloc, default_size, item_size);
     if (ret == AWS_OP_SUCCESS) {
         AWS_POSTCONDITION(aws_priority_queue_is_valid(queue));
+    } else {
+        AWS_POSTCONDITION(aws_priority_queue_is_wiped(queue));
```
Comment (danielsn) [inline source/priority_queue.c]: Change to use macros from https://github.com/awslabs/aws-c-common/pull/453

File: .cbmc-batch/jobs/aws_priority_queue_init_dynamic/aws_priority_queue_init_dynamic_harness.c (hunk 2 above)
Comment (tegansiobhan) [reply .cbmc-batch/jobs/aws_priority_queue_init_dynamic/aws_priority_queue_init_dynamic_harness.c]: Old comment left from earlier debugging. Removed now.

File: source/priority_queue.c (hunk 4 above)
Comment (tegansiobhan) [reply source/priority_queue.c]: Now using the AWS_IS_ZEROED macro

File: include/aws/common/priority_queue.h (hunk 3 above)
Comment (tegansiobhan) [reply include/aws/common/priority_queue.h]: Now using the AWS_IS_ZEROED macro

**Comment by tautschnig** [comment]
@tegansiobhan CBMC reports `**** WARNING: no body for function aws_array_list_is_wiped` -- is this expected, or is there a call left over for a function that should no longer exist?

**Comment by ColdenCullen** [comment]
Looks like this just needs a clang-format run

**Comment by ColdenCullen** [comment]
@danielsn you're review is blocking this, whatchu thinking?

**Pull request merged**